scrowl-io/scrowl
# sc-1043 - Feature - File System Service

### Short summary
Implemented a backend (main) service to handle files on the user's system:
- find file
- open file
- create/save file
---

### Test steps

Remove the `node_modules/`, `apps/**/node_modules`, `.parcel-cache/` folders and run `yarn` to install all fresh poackages.

Run `yarn start:tool:course` and go through the following list:
- [x] Create new project:
1. Click on `New Project`, the `Import Image` and `Save Project` should be enabled. The temporary project folder will be in the console.log.
2. Go to the temporary file using Mac's Finder and you will see a `.project` file. Open it using a text editor and you should see a JSON course structure.
3. Keep the finder open and follow the next step.

- [x] Save (zip) the project
1. Click on `Save Project`. A file dialog should appear. Update the file name and save it.
2. Go to the destination folder and you should see a `.scrowl` file. 
4. Open it with `Archive Utility` and it will extract a folder. That folder should contain a directory with the JSON course file.

- [x] Import an image and copy opened files
1. Click on `Import Image` and a file dialog should pop up allowing you to select image files only.
2. Open the file and you should see the image in an element right below the `Import Image` button.
3. Inspect the image element and the `src` should point to the file copied to the temporary folder.
4. Check out the temporary folder using Mac's finder to ensure the image has been copied.

- [x] Save the updated content to the project file
1. After importing the image, click on `Save Project`. If you saved it once, the dialog should not show up. The same file you saved before will be updated, otherwise, the dialog will appear.
2. Go to the destination folder and you should see a `.scrowl` file. 
3. Rename it to `.zip` and extract its contents. It should contain a directory with the JSON course file and the imported images.
